Prepare 1.6.0 release (#2642)
haskell/haskell-language-server: ChangeLog.md

  - [Qualify imported names](https://haskell-language-server.readthedocs.io/en/latest/features.html#qualify-imported-names) thanks to @eddiemundo
  - New plugin to support [selection range](https://haskell-language-server.readthedocs.io/en/latest/features.html#selection-range) (aka double click text selection) thanks to @kokobd
- Finally hls [supports *ghc 9.2.1*](https://github.com/haskell/haskell-language-server/issues/2179)
  - Including core features and [many plugins](https://haskell-language-server.readthedocs.io/en/latest/supported-versions.html#plugins-support-by-ghc-version))
  - Thanks to a great collective effort coordinated by @pepeiborra and with the help of @wz100, @mpickering and @alanz among others
- Hls now also [supports *ghc 9.0.2*](https://github.com/haskell/haskell-language-server/issues/297) with support for all plugins but the stylish-haskell formatter
- And many many fixes and performance improvement

### Deprecation notice for 1.6.0

- As we noted in the previous release we have dropped support for ghc versions 8.10.5 and 8.8.3 in *this release*

>   - The advise is upgrade ghc to the last minor version: 8.8.4 or 8.10.7

Reviewer: ```suggestion
  - We recommend upgrading ghc to the last minor version: 8.8.4 or 8.10.7
```

  - You can read more about ghc deprecation policy and schedule [here](https://haskell-language-server.readthedocs.io/en/latest/supported-versions.html)
- *After this release*:
  - [We will remove all project stack.yaml's](https://github.com/haskell/haskell-language-server/issues/2533) but two: one for last lts and other for nightly. Temporary we could keep one more stack yaml when nightly upgrades the ghc version, to help in the transition
  - [We will remove the install script](https://github.com/haskell/haskell-language-server/issues/2491) which lives [here](https://github.com/haskell/haskell-language-server/tree/master/install)

>     - The advise is install hls from source using `ghcup`. Download it and run `ghcup compile hls --help` to get moreinfo about.

Reviewer: ```suggestion
    - If you want to install hls from source we recommend using `ghcup`. Download it and run `ghcup compile hls --help` to get more info about.
```

# Changelog for haskell-language-server

## 1.6.0

Time for a new and exciting hls release:

- It includes *three* brand new plugins:
  - [Alternate number literals](https://haskell-language-server.readthedocs.io/en/latest/features.html#convert-numbers-to-alternative-formats) thanks to @drsooch
  - [Qualify imported names](https://haskell-language-server.readthedocs.io/en/latest/features.html#qualify-imported-names) thanks to @eddiemundo
  - New plugin to support [selection range](https://haskell-language-server.readthedocs.io/en/latest/features.html#selection-range) (aka double click text selection) thanks to @kokobd
- Finally hls [supports *ghc 9.2.1*](https://github.com/haskell/haskell-language-server/issues/2179)
  - Including core features and [many plugins](https://haskell-language-server.readthedocs.io/en/latest/supported-versions.html#plugins-support-by-ghc-version))
  - Thanks to a great collective effort coordinated by @pepeiborra and with the help of @wz1000, @mpickering and @alanz among others

> - Hls now also [supports *ghc 9.0.2*](https://github.com/haskell/haskell-language-server/issues/297) with support for all plugins but the stylish-haskell formatter

Reviewer 2: As stylish-haskell remains as a "special" case, maybe it is worth noting that the support is blocked by an upstream issue? Or even better, with a link to the issue: https://github.com/haskell/stylish-haskell/issues/378

Reviewer 2: Also,
```suggestion
- Hls now also [supports *ghc 9.0.2*](https://github.com/haskell/haskell-language-server/issues/297) with all plugins but the stylish-haskell formatter
```
the repetition of "support" sounds a bit redundant.

# Changelog for haskell-language-server

## 1.6.0

Time for a new and exciting hls release:

- It includes *three* brand new plugins:
  - [Alternate number literals](https://haskell-language-server.readthedocs.io/en/latest/features.html#convert-numbers-to-alternative-formats) thanks to @drsooch
  - [Qualify imported names](https://haskell-language-server.readthedocs.io/en/latest/features.html#qualify-imported-names) thanks to @eddiemundo
  - New plugin to support [selection range](https://haskell-language-server.readthedocs.io/en/latest/features.html#selection-range) (aka double click text selection) thanks to @kokobd
- Finally hls [supports *ghc 9.2.1*](https://github.com/haskell/haskell-language-server/issues/2179)
  - Including core features and [many plugins](https://haskell-language-server.readthedocs.io/en/latest/supported-versions.html#plugins-support-by-ghc-version))

>   - Thanks to a great collective effort coordinated by @pepeiborra and with the help of @wz100, @mpickering and @alanz among others

Reviewer 3: ```suggestion
  - Thanks to a great collective effort coordinated by @pepeiborra and with the help of @wz1000, @mpickering and @alanz among others
```

- And many many fixes and performance improvements, thanks to all contributors!

### Deprecation notice for 1.6.0

- As we noted in the previous release we have dropped support for ghc versions 8.10.5 and 8.8.3 in *this release*
  - We recommend upgrading ghc to the last minor version: 8.8.4 or 8.10.7
  - You can read more about ghc deprecation policy and schedule [here](https://haskell-language-server.readthedocs.io/en/latest/supported-versions.html)
- *After this release*:
  - [We will remove all project stack.yaml's](https://github.com/haskell/haskell-language-server/issues/2533) but two: one for last lts and other for nightly. Temporary we could keep one more stack yaml when nightly upgrades the ghc version, to help in the transition
  - [We will remove the install script](https://github.com/haskell/haskell-language-server/issues/2491) which lives [here](https://github.com/haskell/haskell-language-server/tree/master/install)
    - If you want to install hls from source we recommend using `ghcup`. Download it and run `ghcup compile hls --help` to get more info about.

>     - Alternatively users could install hls manually following [these instructions](https://haskell-language-server.readthedocs.io/en/latest/installation.html#installation-from-source)

Reviewer 2: To make this more clear, I think it would be better to add a deprecation note at the beginning of [this doc](https://haskell-language-server.readthedocs.io/en/latest/installation.html#install-specific-ghc-version).
If a user scrolls down to this next section, he/she may get confused.

Author: hmm, the plan would be replace installation instructions removing all references to install script. I thought do it just after the release but you are right if someone follows the link in the meanwhile it will see contradictory assertions

Author: i started to write the new install doc but i dont get it right easily so i think is better to remove the reference for now

- Correctly handle LSP shutdown/exit
([#2486](https://github.com/haskell/haskell-language-server/pull/2486)) by @pepeiborra
- Fix hls-graph ide build with embed-files
([#2485](https://github.com/haskell/haskell-language-server/pull/2485)) by @pepeiborra
- Send unhandled exceptions to the user
([#2484](https://github.com/haskell/haskell-language-server/pull/2484)) by @pepeiborra
- Fix redundant import actions for names starting with _
([#2483](https://github.com/haskell/haskell-language-server/pull/2483)) by @Ailrun
- Update flake to use fourmolu plugin in GHC 9
([#2482](https://github.com/haskell/haskell-language-server/pull/2482)) by @Ailrun
- Delete some dead or deprecated settings
([#2481](https://github.com/haskell/haskell-language-server/pull/2481)) by @michaelpj
- Class plugin bump up
([#2475](https://github.com/haskell/haskell-language-server/pull/2475)) by @Ailrun
- Fix some pragma completion cases
([#2474](https://github.com/haskell/haskell-language-server/pull/2474)) by @Ailrun
- Minor org to contribution doc
([#2472](https://github.com/haskell/haskell-language-server/pull/2472)) by @Anton-Latukha

> - Warn if TH and Mac and static binary

Reviewer 3: Since we have rolled back the Musl change, we should tweak this warning to fire not only in Mac but also in Linux

Author: agree, will do in this pr

Author: done with 539f00, i've relaxed the warning as th evaluation works in some cases for linux
